dot/parachain/collator-protocol/validator_side.go

```diff
@@ -397,6 +778,43 @@ func (f fetchedCollationInfo) String() string {
 		f.relayParent.String(), f.paraID, f.candidateHash.Value.String(), f.collatorID)
 }
 
+func (f *fetchedCollationInfo) FromString(str string) error {
+	splits := strings.Split(str, ",")
+	if len(splits) != 4 {
+		return fmt.Errorf("invalid string format for fetched collation info")
+	}
+
+	relayParent, err := common.HexToHash(strings.TrimSpace(splits[0]))
+	if err != nil {
+		return fmt.Errorf("getting relay parent: %w", err)
+	}
+	f.relayParent = relayParent
+
+	paraID, err := strconv.ParseUint(strings.TrimSpace(splits[1]), 10, 64)
+	if err != nil {
+		return fmt.Errorf("getting para id: %w", err)
+	}
+	f.paraID = parachaintypes.ParaID(paraID)
+
+	candidateHashBytes := common.MustHexToBytes(strings.TrimSpace(splits[2]))
```

Comment:
I would suggest using `HexToBytes` and return an error instead of `MustHexToBytes` that panics the node, also  in the `relayParent` you're using `HexToHash` instead of `MustHexToHash`